iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/tsfile/TsFileManager.java

```diff
@@ -319,12 +322,26 @@ public Set<Long> getTimePartitions() {
     }
   }
 
+  public void initCompaction() {
+    IoTDBConfig config = IoTDBDescriptor.getInstance().getConfig();
+    if (!config.isEnableSeqSpaceCompaction()
+        && !config.isEnableUnseqSpaceCompaction()
+        && !config.isEnableCrossSpaceCompaction()) {
+      return;
+    }
+    CompactionTaskManager.getInstance().register(this);
+  }
+
   public boolean isAllowCompaction() {
     return allowCompaction;
   }
 
   public void setAllowCompaction(boolean allowCompaction) {
     this.allowCompaction = allowCompaction;
+    if (!allowCompaction) {
```

Comment:
What if the `setAllowCompaction(true)` is invoked after the thread pool is shutdown ?